src/TorchVision/dsets/Utils.cs

```diff
@@ -0,0 +1,313 @@
+                {
+                    return md5 == calculate_md5(fpath);
+                }
+
+                internal static bool check_integrity(string fpath, string? md5 = null)
+                {
+                    if (!File.Exists(fpath)) {
+                        return false;
+                    }
+                    if (md5 is null) {
+                        return true;
+                    }
+                    return check_md5(fpath, md5);
+                }
+
+                private static async Task<(string, byte[], Stream)> _extract_gdrive_api_response_async(
+                    HttpResponseMessage response,
+                    int chunk_size = 32 * 1024,
+                    CancellationToken cancellationToken = default)
+                {
+                    var stream = await response.Content.ReadAsStreamAsync();
+                    var buf = new byte[chunk_size];
+                    int pos = 0;
+                    while (pos < chunk_size) {
+                        int readLen = await stream.ReadAsync(buf, pos, chunk_size - pos, cancellationToken);
+                        if (readLen == 0) break;
+                        pos += readLen;
+                    }
+                    buf = buf.AsSpan(0, pos).ToArray();
+
+                    string api_response = string.Empty;
+                    string first_chunk = Encoding.ASCII.GetString(buf);
+                    string pattern = "<title>Google Drive - (?<api_response>.+?)</title>";
+                    var match = Regex.Match(first_chunk, pattern);
+                    if (match.Success) {
+                        api_response = match.Groups["api_response"].Value;
+                    }
+                    return (api_response, buf, stream);
+                }
+
+                /// <summary>
+                /// Download a Google Drive file and place it in root.
+                /// </summary>
+                /// <param name="file_id">id of file to be downloaded</param>
+                /// <param name="root">Directory to place downloaded file in</param>
+                /// <param name="filename">Name to save the file under. If null, use the id of the file.</param>
+                /// <param name="md5">MD5 checksum of the download. If null, do not check</param>
+                public static void download_file_from_google_drive(
+                    string file_id, string root, string? filename = null, string? md5 = null)
+                {
+                    try {
+                        Task.Run(async () => {
+                            await download_file_from_google_drive_async(file_id, root, filename, md5);
+                        }).Wait();
+                    } catch (AggregateException ex) {
+                        throw ex.InnerException ?? ex;
+                    }
+                }
+
+                internal static async Task download_file_from_google_drive_async(
```

Comment:
This sounds broadly useful, so should probably be in an IO namespace, have a .NET-convention name, and be public...

Reply:
> be in an IO namespace

Moved the functionality to `torchvision.io` static class. I guess you didn't mean .NET `TorchSharp.IO` namespace.